Rebus.AdoNet/AdoNetTimeoutStorage.cs

```diff
@@ -80,22 +80,65 @@ public void Add(Timeout.Timeout newTimeout)
 			}
 		}
 
+		public string GetSqlForDialect(IDbConnection connection)
+		{
+			// PostgresSql Dialects
+			if (factory.Dialect.SupportsPgTryAdvisoryXactLock || factory.Dialect.SupportsPgTryAdvisoryLock || factory.Dialect.SupportsSkipLocked)
+			{
+				return GetSqlForPostgreSqlDialect(new Version(factory.Dialect.GetDatabaseVersion(connection)));
```

Comment:
Dialect has to be received/guessed during initialization, instead of doing it on every iteration.